File: touch/src/main/java/de/nicidienase/chaosflix/touch/browse/adapters/ConferenceGroupsFragmentPager.java
package de.nicidienase.chaosflix.touch.browse.adapters;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.util.Log;

import java.util.ArrayList;
import java.util.List;

import de.nicidienase.chaosflix.touch.R;
import de.nicidienase.chaosflix.common.mediadata.entities.recording.persistence.ConferenceGroup;
import de.nicidienase.chaosflix.touch.browse.ConferenceGroupFragment;

public class ConferenceGroupsFragmentPager extends FragmentPagerAdapter {

	private static final String TAG = ConferenceGroupsFragmentPager.class.getSimpleName();
	private final Context mContext;
	private List<ConferenceGroup> conferenceGroupList = new ArrayList<>();

	public ConferenceGroupsFragmentPager(Context context, FragmentManager fm) {
		super(fm);
		this.mContext = context;
	}

	@Override
	public Fragment getItem(int position) {
		ConferenceGroup conferenceGroup = conferenceGroupList.get(position);
		ConferenceGroupFragment conferenceFragment = ConferenceGroupFragment.newInstance(conferenceGroup, getNumColumns());
		Log.d(TAG, "Created Fragment for: " + conferenceGroup.getName());
		return conferenceFragment;
	}

	@Override
	public long getItemId(int position) {
		return conferenceGroupList.get(position).getId();
	}

	@Override
	public int getCount() {
		return conferenceGroupList.size();
	}

	@Override
	public CharSequence getPageTitle(int position) {
		return conferenceGroupList.get(position).getName();
	}

	private int getNumColumns() {
		return mContext.getResources().getInteger(R.integer.num_columns);
	}

	public void addGroup(ConferenceGroup conferenceGroup) {
		conferenceGroupList.add(conferenceGroup);
		//		Collections.sort(conferenceGroupList,(g1, g2) -> g1.getIndex()-g2.getIndex());
		notifyDataSetChanged();
	}

	public void setContent(List<ConferenceGroup> conferenceGroups) {
		conferenceGroupList = conferenceGroups;
		notifyDataSetChanged();
	}
}

File: touch/src/main/java/de/nicidienase/chaosflix/touch/browse/adapters/ConferenceGroupsFragmentPager.kt
package de.nicidienase.chaosflix.touch.browse.adapters

import android.content.Context
import android.util.Log
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import de.nicidienase.chaosflix.common.mediadata.entities.recording.persistence.ConferenceGroup
import de.nicidienase.chaosflix.touch.R
import de.nicidienase.chaosflix.touch.browse.ConferenceGroupFragment.Companion.newInstance
import java.util.ArrayList

class ConferenceGroupsFragmentPager(private val context: Context, fm: FragmentManager?) : FragmentPagerAdapter(fm!!) {

    private var conferenceGroupList: MutableList<ConferenceGroup> = ArrayList()

    override fun getItem(position: Int): Fragment {
        val conferenceGroup = conferenceGroupList[position]
        val conferenceFragment = newInstance(conferenceGroup, numColumns)
        Log.d(TAG, "Created Fragment for: " + conferenceGroup.name)
        return conferenceFragment
    }

    override fun getItemId(position: Int): Long {
        return conferenceGroupList[position].id
    }

    override fun getCount(): Int {
        return conferenceGroupList.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return conferenceGroupList[position].name
    }

    private val numColumns: Int by lazy { context.resources.getInteger(R.integer.num_columns) }

    fun setContent(conferenceGroups: MutableList<ConferenceGroup>) {
        conferenceGroupList = conferenceGroups
        notifyDataSetChanged()
    }

    companion object {
        private val TAG = ConferenceGroupsFragmentPager::class.java.simpleName
    }
}
